File: food-ordering-system/src/main/java/com/foodorder/service/OrderService.java
package com.foodorder.food_ordering_system.service;

import com.foodorder.food_ordering_system.dto.OrderDto;
import java.util.List;

public interface OrderService {
    OrderDto placeOrder(Long userId);
    OrderDto updateOrderStatus(Long orderId, String status);
    List<OrderDto> getOrdersByUser(Long userId);
    List<OrderDto> getAllOrders();
}

File: food-ordering-frontend/src/api/orderService.ts
import api from './axios';

export interface OrderItem {
  foodItemId: number;
  foodItemName: string;
  quantity: number;
  price: number;
}

export interface Order {
  id: number;
  userId: number;
  status: string;
  orderDate: string;
  totalAmount: number;
  orderItems: OrderItem[];
}

export const orderService = {
  placeOrder: () => api.post<Order>('/orders/place'),
  getMyOrders: () => api.get<Order[]>('/orders/my-orders'),
  getOrderById: (id: number) => api.get<Order>(`/orders/${id}`),
  getAllOrders: () => api.get<Order[]>('/orders/all'),
  updateOrderStatus: (id: number, status: string) =>
    api.put<Order>(`/orders/${id}/status`, null, { params: { status } }),
};